src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,56 @@
 #include "chunk.hpp"
+#include <boost/hana/for_each.hpp>
 
 #include "abstract_segment.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  DebugAssert(_segments.size() < std::numeric_limits<ColumnCount>::max(), "Segment limit is already reached.");
+  _segments.emplace_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  auto size = _segments.size();
+  DebugAssert(values.size() == size, "Number of values and number of columns should be equal.");
+
+  for (size_t i = 0; i < size; ++i) {
+    const auto& value = values.at(i);
+    const auto segment = _segments.at(i);
+
+    // If our value is not a NullValue, we could infer from value.type() which type of ValueSegment we need
+    // to cast our segment to in order to be able to append values to it.
+    // However, if the value is NullValue, we have no way to infer this. Therefore, we have no choice but to
+    // just try everything until we find something that works...
+    bool append_successful = false;
```

Comment:
Here we could also use the above style: 
```suggestion
    auto append_successful = false;
```